sys/riscv/include/pte.h

```diff
@@ -70,8 +70,9 @@ typedef	uint64_t	pn_t;			/* page number */
 /* CHERI uses reserved bits in 55:63 */
 #define	PTE_CW		(1UL << 63) /* Capability Write */
 #define	PTE_CR		(1UL << 62) /* Capability Read */
-#define	PTE_KERN_CHERI	(PTE_CR | PTE_CW)
-#define	PTE_PROMOTE_CHERI (PTE_CR | PTE_CW)
+#define	PTE_CD		(1UL << 61) /* Capability Dirty */
+#define	PTE_KERN_CHERI	(PTE_CR | PTE_CW | PTE_CD)
+#define	PTE_PROMOTE_CHERI (PTE_CR | PTE_CW | PTE_CD)
```

Comment:
AFAICT `PTE_PROMOTE` is used in two different senses: to decide when the `pmap` entry must be stored back as part of demoting a superpage back to ordinary pages and to gate superpage formation from ordinary pages.  `PTE_CD` is not necessary in the second case; that is, it need not be the case that `PTE_CD` be set on all L3 pages to make an L2 page with `PTE_CD` set.  (Though there's no real harm; if you want to land this as a baseline and do the fine tuning later, or never, that's also probably fine, but a breadcrumb comment is probably useful even so.)

Reply:
The normal PTE_PROMOTE has PTE_D in it. Plus any distinction is irrelevant given CD and CW are kept in sync here.

Reply:
Fair 'nough.  In that case, LGTM.